### PR TITLE
Jspurlin/swatch color picker simplify implementation

### PR DESCRIPTION
**Pull request by jspurlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Simplify the SwatchColorPicker to not contain the ability to render inside of a menu button or know about header/divider options. This component will now only render the grid aspect

#### Focus areas to test

Updated unit tests, verified focus, general keyboarding, rendering and execution works as expected

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx [hunk 1]
```diff
@@ -89,8 +76,6 @@ const SwatchColorPickerOption: React.StatelessComponent<ISwatchColorPickerOption
     let {
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx]: Something I learned in fabric-internal is that if you are doing logic with your callbacks and you need to create separate functions like we are doing here, we shouldn't use StatelessComponent.

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts [hunk 2]
```diff
@@ -29,18 +32,6 @@ export interface ISwatchColorPickerProps {
...
-  /**
-   * Should the icon color be updated to align
-   * with the selected color? Default is false
-   */
-  setSelectedColorForIcon?: boolean;
-
-  /**
    * The color options that will be made available to the user
    */
   swatchColorPickerItems: ISwatchColorPickerItemProps[];
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts]: should we take the opportunity to rename swatchColorPickerItems to colorCells?

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts [hunk 3]
```diff
@@ -19,6 +17,11 @@ export interface ISwatchColorPickerProps {
   id?: string;
 
   /**
+   * Additional class name to provide on the root element
+   */
+  className?: string;
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts]: should we glamorize this component instead or is that a separate PR?

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts [hunk 4]
```diff
@@ -105,20 +106,7 @@ export interface ISwatchColorPickerItemProps {
...
   /**
-   * The menu item button props. This value is only used if
-   * the type is MenuItem
-   */
-  menuItemButtonProps?: IButtonProps;
-
-  /**
    * Is this individual item disabled?
    */
   disabled?: boolean;
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts]: Does this make sense in the context of a color cell grid?

**Comment by jair-cazarin** [comment]
Should this go to master or 5.0 since it's breaking the entire contract?

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts (hunk 3 above)
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts]: That would be a separate PR. Adding this overriding className is a piece of feedback from David

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts (hunk 4 above)
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.Props.ts]: I don't see why not, although I'm not aware of any instance where this happens currently in our implementation... I'll remove it and it could always be added back if we find a case where it's needed

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx (hunk 1 above)
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx]: Cool, I'll convert this from a StatelessComponent to a React.Component

**Comment by jspurlin** [comment]
@jair-cazarin This should go into the next version of master that is supposed to be pushed out soon. @dzearing as an FYI
